Lib/test/test_signal.py

```diff
@@ -519,17 +519,25 @@ def handler(signum, frame):
         else:
             write.setblocking(False)
 
-        # Start with large chunk size to reduce the
-        # number of send needed to fill the buffer.
         written = 0
-        for chunk_size in (2 ** 16, 2 ** 8, 1):
-            chunk = b"x" * chunk_size
+        if sys.platform == "vxworks":
             try:
                 while True:
-                    write.send(chunk)
-                    written += chunk_size
-            except (BlockingIOError, socket.timeout):
+                    write.send(b"x")
+                    written += 1
+            except BlockingIOError:
                 pass
+        else:
+            # Start with large chunk size to reduce the
+            # number of send needed to fill the buffer.
+            for chunk_size in (2 ** 16, 2 ** 8, 1):
```

Comment:
I would prefer to not copy/paste the code, restrict the difference to the loop. Something like:

```suggestion
            if sys.platform == "vxworks": 
                CHUNK_SIZES = (1,)
            else:
                CHUNK_SIZES = (2 ** 16, 2 ** 8, 1)
            for chunk_size in CHUNK_SIZES:
```

Reply:
Better solution. Changed.